recirq/benchmarks/rep_rate/rep_rate_calculator.py

```diff
@@ -0,0 +1,263 @@
+                m = m.with_operation(cirq.X(q)**sympy.Symbol(symbol))
+            else:
+                m = m.with_operation(
+                    cirq.PhasedXZGate(x_exponent=random.random(),
+                                      z_exponent=random.random(),
+                                      axis_phase_exponent=random.random())(q))
+        return (m, current_sym)
+
+    def _create_rep_rate_circuit(
+            self, parameterized: bool, width: int, depth: int,
+            num_sweeps: Optional[int]) -> Tuple[cirq.Circuit, cirq.Sweep]:
+        """Creates a testing circuit based on parameters.
+
+        The circuit will be of alternating single and two qubit layers
+        using a number of qubits specified by width and a number of
+        moments specified by depth.
+
+        This function will also create a sweep if parameterized is true,
+        by parameterized all single qubit layers with random values for
+        angles.
+        """
+        qubits = self.qubits[:width]
+        c = cirq.Circuit()
+        symbol_count = 0
+        for layer in range(depth):
+            if layer % 2:
+                moment, symbol_count = self._sq_layer(qubits, parameterized,
+                                                      symbol_count)
+                c.append(moment)
+            else:
+                c.append(self._entangling_layer(qubits))
+        c.append(cirq.Moment(cirq.measure(*qubits)))
+
+        if not parameterized:
+            return (c, None)
+        sweeps = cirq.Zip(*[
+            cirq.Linspace('s_%d' % i, start=0, stop=1, length=num_sweeps)
+            for i in range(symbol_count)
+        ])
+        return (c, sweeps)
+
+    def get_samples(self, circuit: cirq.Circuit,
+                    num_trials: int) -> List[float]:
+        """Use a sampler to execute the circuit num_trial times.
+
+        Returns:
+           a list of durations, in seconds for the executions.
+        """
+        latencies = []
+        for trial in range(num_trials):
+            self._print(f'Executing Trial {trial}')
+            start = time.time()
+            self.sampler.run(circuit, repetitions=100)
+            end = time.time()
+            duration = end - start
+            latencies.append(duration)
+            self._print(f'Finished Trial {trial} with duration {duration}')
+        return latencies
+
+    def get_latency_samples(self, num_trials: int) -> List[float]:
+        return self.get_samples(self._latency_circuit(), num_trials)
+
+    def print_latency(self, num_trials: int) -> None:
+        self._flush_print_log()
+        latencies = self.get_latency_samples(num_trials)
+        latencies = sorted(latencies)
+        median = latencies[len(latencies) // 2]
+        p05 = latencies[len(latencies) // 20]
+        p95 = latencies[-len(latencies) // 20]
+        p10 = latencies[len(latencies) // 20]
+        p90 = latencies[-len(latencies) // 20]
+        average = sum(latencies) / len(latencies)
```

Comment:
might be less error prone and faster to use `np.median` `np.percentile`

Reply:
Done.